ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/TrailingCommaOnDeclarationSiteRule.kt

```diff
@@ -327,8 +329,8 @@ public class TrailingCommaOnDeclarationSiteRule :
                                 prevNode
                                     .treeParent
                                     .indent()
-                            if (leafBeforeArrowOrNull is PsiWhiteSpace) {
-                                (leafBeforeArrowOrNull as LeafPsiElement).rawReplaceWithText(indent)
+                            if (leafBeforeArrowOrNull.elementType == ElementType.WHITE_SPACE) {
+                                (leafBeforeArrowOrNull.psi as LeafPsiElement).rawReplaceWithText(indent)
```

Comment:
Replace with:
```
                            if (leafBeforeArrowOrNull.isWhiteSpace()) {
                                (leafBeforeArrowOrNull?.psi as LeafPsiElement).rawReplaceWithText(indent)
```

Reply:
The second line was not changed. 

My IDEA shows an error:
<img width="1165" alt="Screenshot 2024-12-22 at 14 25 12" src="https://github.com/user-attachments/assets/983b6325-41bb-4ffe-88a9-12f41e71df9f" />
Strangely enough it can be compiled successfully, and all tests succeed. But still I like not to see any errors ;-)

We can make it even betters, as we do need to convert to psi:
```
                                (leafBeforeArrowOrNull as LeafElement).rawReplaceWithText(indent)
```

Reply:
> [ASTNode](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/lang/ASTNode.java) doesn't have a @Nullable annotation for getPsi(), but it does for other members. Maybe K2 interprets the lack of @Nullable to mean that it it is not nullable.

K2 not enabled on my machine. Get it both with Ultimate versions 2024.2 and 2024.3. But it is not important anymore as the `.psi` conversion is removed.

> Implemented with getPsi as you suggested

I guess you meant that you have removed the conversion to `psi` as requested.